### PR TITLE
Update changelog and add pull request links

### PR DESCRIPTION
**Pull request by mathroule**
## Description

Update changelog and add pull request links in changelog.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (non-breaking change that does not add functionality but updates documentation)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/markdownlint/markdownlint/blob/main/CONTRIBUTING.md) document.
- [x] Wrote [good commit messages](https://chris.beams.io/posts/git-commit/)
- [x] Feature branch is up-to-date with `main`, if not - rebase it
- [x] Added tests for all new/changed functionality, including tests for positive and negative scenarios
- [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,33 +2,37 @@
 
 ## Unreleased
 
-## [v0.13.0] (2023-10-01)
```
Comment (jaymzh) [inline CHANGELOG.md:5]: you dropped the version header, you need to put that back.

**Review by jaymzh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mathroule) [reply CHANGELOG.md:5]: Oops! Thanks for catching that ;)
It's now reverted.

**Review by jaymzh — APPROVED** [review]

**Review by joe10832 — APPROVED** [review]
@markdownlint-autobuild

File: CHANGELOG.md (hunk 1 above)
Comment (joe10832) [reply CHANGELOG.md:5]: @GEMINI.md

**Pull request merged**